Add migration docs (#9)
exercism/v3-launch: migration/build-test-runner.md

# Build Test Runner

In Exercism V3, one of the biggest changes is that we'll automatically check if a submitted solution passes all the tests.

> We'll check this via a new, track-specific tool: the [_test runner_](https://github.com/exercism/v3-docs/tree/master/anatomy/track-tooling/test-runners). The test runner is track-specific and implemented as a Docker image. Whenever a solution is submitted, a Docker container is created of that image and it runs on the submitted solution. The test runner outputs a JSON file that describes the test results.

Reviewer: I worry a little that we scare people by mentioning Docker here. The Docker bit is actually a small/easy bit that we can help with, but for lots of people it might be offputting.

```suggestion
We'll check this via a new, track-specific tool: the [_test runner_](https://github.com/exercism/v3-docs/tree/master/anatomy/track-tooling/test-runners). Each test runner is track-specific. When a new solution is submitted, we run the track's test-runner, which outputs a JSON file that describes the test results.

```

Author: Fair point

The test runner must be able to run the tests suites of both [Concept Exercises](https://github.com/exercism/v3-docs/blob/master/product/concept-exercises.md) and [Practice Exercises](https://github.com/exercism/v3-docs/blob/master/product/practice-exercises.md).

> Each track must build a test runner according to [the spec](https://github.com/exercism/v3-docs/blob/master/anatomy/track-tooling/test-runners/interface.md).

Reviewer: We should ensure that we recommend people use the `generic-test-runner` as a starting point in the docs somewhere.
